ui/src/ImageForm/components/ImageForm.js

```diff
@@ -0,0 +1,156 @@
+        console.log("Fetching configmap from images")
+        fetch('/services/jsp-api/api/user/configmap', {method:'GET'})
+            .then(response => {
+                if (response.ok) {
+                    return response.json();
+                } 
+                else {
+                    throw new Error('Failed to fetch user cm');
+                }
+            })
+            .then(data => {
+                console.log("Received image: ", data['last_selected_image'])
+                //this.setState({selectedValue: data['last_selected_image']}, this.isImageEmpty())
+                this.state.selectedValue = data['last_selected_image']
+                this.setState({selectedValue: this.state.selectedValue})
+                this.isImageCorrect()
+            }) 
+    }
+
+    updateImages(){
+        fetch('/services/jsp-api/api/images', {method: 'GET'})
+            .then(response => {
+                if (response.ok) {
+                    return response.json();
+                } 
+                else {
+                    throw new Error('Unknown error.');
+                }
+            })
+            .then(data => {
+                this.setState({imageList: data}, this.updateConfigmap());
+            })
+    }
+
+    componentDidMount() {
+        //this.updateConfigmap()
+        this.updateImages()
+        //this.sendInitialImage()
+    }
+
+    getValue(value) {
+        return value
+    }
+
+    isImageCorrect() {
+        console.log("Entered image check function: ", this.state.selectedValue)
+        console.log(this.state.sizeList.find(this.getValue))
+        if (this.state.imageList.find(this.getValue) === undefined) {
+            if (this.state.imageList.length !== 0) {
+                this.setState({selectedValue: this.state.imageList[0]}, console.log("Set default image"))
+                //this.state.selectedValue = this.state.imageList[0]
+                this.postChange(this.state.selectedValue)
+            }
+            else {
+                this.updateImages()
+            }
+        }
+    }
+
+    /*sendInitialImage() {
+        console.log("Configmap empty, sending initial image", this.state.imageList)
+        if (this.state.imageList.length > 0) {
+            var image = this.state.imageList[0]
+            console.log(image)
+        }
+        var json = JSON.stringify({last_selected_image: image})
+        fetch('/services/jsp-api/api/user/configmap',
+            {
+                method: 'POST',
+                body: json,
+                headers:{
+                  'Content-Type': 'application/json',
+                }
+            }
+            )
+        console.log("Sent chosen image:", json);
+        this.updateConfigmap()
+    }*/
+
+    postChange(text) {
+        if (typeof text !== "string") {
+            text = text.target.text
+        }
+        var json = JSON.stringify({last_selected_image: text})
+        fetch('/services/jsp-api/api/user/configmap',
```

Comment:
Can you first setState and then post the CM update?

Right now when I select a different image (or size) it takes a bit of time to update the actual selection - it is annoying:)

Maybe if you first setState with the new value and then ran the `POST`, it would change immediately?

(also, do you need to call `updateConfigmap` after the `POST`?